src/staking/TLCHook.sol

```diff
@@ -55,18 +65,26 @@ contract TLCHook is ITradeServiceHook, OwnableUpgradeable {
     // Do nothing
   }
 
-  function _mintTLC(address _primaryAccount, uint256 _sizeDelta) internal {
+  function _mintTLC(address _primaryAccount, uint256 _sizeDelta, uint256 _marketIndex) internal {
     // SLOADs
     TraderLoyaltyCredit _tlc = TraderLoyaltyCredit(tlc);
     TLCStaking _tlcStaking = TLCStaking(tlcStaking);
     // Calculate mint amount which is equal to sizeDelta but convert decimal from 1e30 to 1e18
-    // This is to make the TLC token composable as ERC20 with regular 18 decimals
-    uint256 _mintAmount = _sizeDelta / 1e12;
+    // This is to make the TLC token composable as ERC20 with regular 18 decimals, also wighted
+    uint256 weight = marketWeights[_marketIndex] == 0 ? BPS : marketWeights[_marketIndex];
+    uint256 _mintAmount = _sizeDelta.mulDiv(weight, 1e12) / BPS;
```

Comment:
can optimized math here. Instead of `mulDiv(weight, 1e12)` you can do `mulDiv(weight, 1e12 * BPS)`